2.segment_images/scripts/2.segment_whole_organoids.py

```diff
@@ -40,7 +40,64 @@
 print(in_notebook)
 
 
-# In[6]:
+# In[2]:
+
+
+def segment_with_diameter(
+    img: np.ndarray,
+    model: models.Cellpose,
+    diameter: int,
+    z_axis: int = 0,
+    channels: tuple = [1, 0],
+):
+    """
+    Recursively perform segmentation, stepping down through diameters by 250
+    until a valid label is found or the minimum diameter is reached.
+    This effectively performs a dynamic search for the largest detectable object
+    in the image.
+
+    Parameters
+    ----------
+    img : np.ndarray
+        The image to segment. Can be 3D in the format of (z, y, x).
+    model : models.Cellpose
+        The Cellpose model to use for segmentation.
+    diameter : int
+        The diameter to use for segmentation.
+        This is where the search starts.
+    z_axis : int, optional
+        The axis of the z-stack. Default is axis 0.
+    channels : tuple, optional
+        The channels to use for segmentation. Default is (1, 0).
+        Where 1 is the channel for the cytoplasm and 0 using no other channel.
+
+    Returns
+    -------
+    tuple
+        labels : np.ndarray
+            The labels of the segmented image.
+        details : dict
+            The details of the segmentation.
+        _ : None
```

Comment:
Consider removing this value and relying instead on a Dict. This would allow you to expand the work through a type without getting caught in a specific number of return values.